src/hotspot/share/cds/metaspaceShared.cpp

```diff
@@ -670,9 +670,11 @@ void MetaspaceShared::preload_and_dump(TRAPS) {
     }
   }
 
-  if (AOTMode != nullptr && strcmp(AOTMode, "create") == 0) {
-    // We can't return to the JLI launcher, as it will try to run the main class, but
-    // the main class is not specified when -XX:AOTMode=create is used.
+  if (!CDSConfig::old_cds_flags_used()) {
+    // The JLI launcher only recognizes the "old" -Xshare:dump flag.
+    // When the new -XX:AOTMode=create flag is used, we can't return
+    // to the JLI launcher, as the launcher will fail when trying to
+    // run the main class, which is not what we want.
```

Comment:
I actually find the original more clear - we have a new flag that we check and act upon. The "!old flag" seems an awkward way of saying "new flag".